src/main/java/org/dice_research/opal/civet/metrics/ContactClassicMetric.java

```diff
@@ -0,0 +1,145 @@
+import org.apache.jena.rdf.model.SimpleSelector;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.rdf.model.StmtIterator;
+import org.apache.jena.sparql.vocabulary.FOAF;
+import org.apache.jena.vocabulary.DCAT;
+import org.apache.jena.vocabulary.DCTerms;
+import org.apache.jena.vocabulary.RDF;
+import org.apache.jena.vocabulary.VCARD;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dice_research.opal.civet.Metric;
+import org.dice_research.opal.common.vocabulary.Opal;
+
+/**
+ * The ContactClassicMetric awards stars based on the availability 
+ * of classical contact details in the dataset like name, address and telephone number.
+ * 
+ * @author Amit Kumar
+ */
+
+public class ContactClassicMetric implements Metric {
+
+	private static final Logger LOGGER = LogManager.getLogger();
+	private static final String DESCRIPTION = "Computes the quality of dataset as per contactability criteria."
+			+ "Four kinds of ratings are awarded to the dataset which are following: "
+			+ "Stars 5: Name, Telephone and Address is found."
+			+ "Stars 4: Any two of the Name, Telephone and Address is found."
+			+ "Stars 2: Any one of the Name, Telephone and Address is in the dataset."
+			+ "Stars 0: Nothing is found"; 
+
+	@Override
+	public Integer compute(Model model, String datasetUri) throws Exception {
+		//creating a dataset
+		LOGGER.info("Processing dataset " + datasetUri);
+		Resource dataset = ResourceFactory.createResource(datasetUri);
+		
+		StmtIterator stmtItr = model.listStatements(new SimpleSelector
+				(dataset,DCAT.contactPoint,(RDFNode) null));
+		NodeIterator publisherObjectsIterator = model.listObjectsOfProperty(dataset,DCTerms.publisher);
+
+		boolean nameFound=false;
+		boolean nameFound2=false;
+		boolean adrFound=false;
+		boolean telephoneFound=false;
+		
+		while(publisherObjectsIterator.hasNext()) {	
+			RDFNode Publisher = publisherObjectsIterator.next();
+			Resource PublisherBlankNode = (Resource) Publisher;
+			if(PublisherBlankNode.hasProperty(FOAF.name)) {
+				if (PublisherBlankNode.getProperty(FOAF.name).getObject() != null) {
+					nameFound = true;
+					}
+				}
+			else if(PublisherBlankNode.hasProperty(FOAF.firstName)||PublisherBlankNode.hasProperty(FOAF.familyName)
+						||PublisherBlankNode.hasProperty(FOAF.lastName)){
+					nameFound = true;										
+				}
+			if (PublisherBlankNode.hasProperty(FOAF.phone)) {
+					if (PublisherBlankNode.getProperty(FOAF.phone).getObject() != null) {
+						telephoneFound = true;					
+						}
+				}
+		}
+		
+		Property customName = model.createProperty("http://xmlns.com/foaf/0.1/name");
+		Property customName2 = model.createProperty("http://www.w3.org/2006/vcard/ns#fn");
+		Property customAdr = model.createProperty("http://www.w3.org/2006/vcard/ns/ns#hasAddress");
+		Property customTelephone = model.createProperty("http://www.w3.org/2006/vcard/ns#hasTelephone");
+
+		
+		if(stmtItr.hasNext())
```

Comment:
We can directly check for DCAT.ContactPoint's object. No need to use an Iterator.